pkg/cli/upgradeassistant/cmd/migrate/1170.go

```diff
@@ -138,3 +146,138 @@ func migrateJiraInfo() error {
 		JiraToken: info.AccessToken,
 	})
 }
+
+func migrateSharedService() error {
+	type ServiceInfo struct {
+		ServiceName string
+		ProductName string
+		Deleted     bool
+	}
+
+	sharedServiceListInEnvMap := map[string]map[string]*ServiceInfo{}
+	sharedServiceInfoList := map[string]*ServiceInfo{}
+
+	// get all shared service in product template
+	productTemplates, err := template.NewProductColl().ListWithOption(
+		&template.ProductListOpt{},
+	)
+	if err != nil {
+		return errors.Wrap(err, "list product templates")
+	}
+	for _, productTemplate := range productTemplates {
+		for _, sharedService := range productTemplate.SharedServices {
+			if _, ok := sharedServiceListInEnvMap[productTemplate.ProductName]; !ok {
+				sharedServiceListInEnvMap[productTemplate.ProductName] = make(map[string]*ServiceInfo)
+			}
+			s := &ServiceInfo{
+				ServiceName: sharedService.Name,
+				ProductName: sharedService.Owner,
+			}
+			sharedServiceListInEnvMap[productTemplate.ProductName][sharedService.Name] = s
+			sharedServiceInfoList[s.ProductName+"-"+s.ServiceName] = s
+		}
+	}
+
+	// get all shared service in product env
+	// if shared service in product env, but not in product template, should mark it as deleted
+	allProductEnvs, err := mongodb.NewProductColl().List(nil)
```

Comment:
i suggest add some filter instead of querying all products since shared services only exist in k8s projects